flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala

```diff
@@ -674,6 +675,35 @@ class TemporalTypesTest extends ExpressionTestBase {
     //testSqlApi("CEIL(TIMESTAMP '2018-03-20 06:10:31' TO HOUR)", "2018-03-20 07:00:00.000")
   }
 
+  @Test
+  def testLocalDateTimeInUTC(): Unit = {
+    config.setLocalTimeZone(ZoneId.of("UTC"))
+
+    val formattedLocalDateTime = LocalDateTime
+      .now(ZoneId.of("UTC"))
+      .format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+
+    // the LOCALTIMESTAMP function is not deterministic, thus we
+    // use following pattern to check it return SQL timestamp in session time zone UTC
+    testSqlApi(
+      s"TIMESTAMPDIFF(MINUTE, TIMESTAMP '$formattedLocalDateTime', LOCALTIMESTAMP) <= 1",
+      "true")
+  }
+
+  @Test
+  def testLocalDateTimeInShanghai(): Unit = {
```

Comment:
Could you add these tests into `NonDeterministicTests#testLocalTimestamp`? 
Besides, we may also need to test `LOCALTIME`?

Reply:
okay，`LOCALTIME` may cross different date eg 23:59:59:000, 00:00:00.001 and I didn't find a time function to compare them, but `LOCALTIME` comes from `LOCALTIMESTAMP` which has cover the time zone code path.